chain-core/src/state/account.rs

```diff
@@ -0,0 +1,115 @@
+use crate::common::{hash256, Timespec, HASH_SIZE_256};
+use crate::init::address::RedeemAddress;
+use crate::init::coin::Coin;
+use blake2::Blake2s;
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+use std::mem;
+
+/// account state update counter
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Copy)]
+pub struct Nonce(u64);
+
+impl From<u64> for Nonce {
+    fn from(v: u64) -> Self {
+        Nonce(v)
+    }
+}
+
+impl From<Nonce> for u64 {
+    fn from(bh: Nonce) -> u64 {
+        bh.0
+    }
+}
+
+impl Encodable for Nonce {
+    fn rlp_append(&self, s: &mut RlpStream) {
```

Comment:
`Encodable` and `Decodable` implementations can be modified similar to `Count` in `chain-abci/src/storage/account/tree.rs`.

Reply:
I'll make a macro

Reply:
probably a derive macro would be good